types/src/key.rs

```diff
@@ -221,6 +224,8 @@ pub enum Key {
     Message(MessageAddr),
     /// A `Key` under which a single named key entry is stored.
     NamedKey(NamedKeyAddr),
+    /// A `Key` under which the total number of emitted messages in a block is stored.
+    BlockMessageCount(BlockTime),
```

Comment:
i have some concerns about the address-ability of this...i'd like to discuss it w/ you in realtime when we can sync up